python/ray/data/impl/arrow_block.py

```diff
@@ -200,6 +200,30 @@ def num_rows(self) -> int:
     def size_bytes(self) -> int:
         return self._table.nbytes
 
+    def zip(self, other: "Block[T]") -> "Block[T]":
```

Comment:
worth document the behavior of zipping two blocks with duplicate columns: undefined/overwrites/etc?